Java/core/src/main/java/com/amazon/randomcutforest/executor/IUpdateCoordinator.java

```diff
@@ -47,9 +44,18 @@
      * further processing if needed.
      * 
      * @param updateResults A list of points that were deleted.
+     * @param updateInput The corresponding output from {@link #initUpdate}, which was passed into
+     *                    the update method for each component model.
      */
     void completeUpdate(List<Optional<UpdateReturn<P>>> updateResults, P updateInput);
 
-    PointStoreDoubleState getPointStoreState();
+    long getTotalUpdates();
+
+    // TODO we can remove this from the interface once this value is being set in
+    // the appropriate mapper
+    void setTotalUpdates(long totalUpdates);
 
+    default long getCurrentIndex() {
```

Comment:
question. does index need to be 1-based?

Reply:
I think we've gone back and forth on this in the 2.0 code. I don't have a strong opinion about it, but this PR does have an opinion about who should own counting updates and keeping track of the current index (namely, the update coordinator). If we're fine with being 0-based, I can remove this method entirely.

The reason I put a +1 there, is because this value used to be maintained in the update executor, and the first point submitted to the forest would get an index of 1 (https://github.com/aws/random-cut-forest-by-aws/blob/compact-rcf/Java/core/src/main/java/com/amazon/randomcutforest/executor/AbstractForestUpdateExecutor.java#L80).

Reply:
Agreed, I'll change this.